File: frontend/src/types/MapMarker.ts
import {Visibility} from "../data/Visibility.ts";
import {MapMarkerStatus} from "../data/MapMarkerStatus.ts";

export type MapMarker = {
    id: string,
    worldMapId: string,
    name: string,
    xPosition: number,
    yPosition: number,
    markerTypeId: string,
    playerArticleId: string,
    gmArticleId: string,
    status: MapMarkerStatus,
    visibility: Visibility
};

export const emptyMapMarker:MapMarker = {
    id: '',
    name: '',
    xPosition: 0,
    yPosition: 0,
    worldMapId: '',
    markerTypeId: '',
    playerArticleId: '',
    gmArticleId: '',
    status: 'ACTIVE',
    visibility: 'OWNER_ONLY'
};

File: backend/src/main/java/com/github/mahambach/backend/model/MapMarker.java
package com.github.mahambach.backend.model;

import com.github.mahambach.backend.model.enums.MapMarkerStatus;
import com.github.mahambach.backend.model.enums.Visibility;
import lombok.With;
import org.springframework.data.annotation.Id;

@With
public record MapMarker(
        @Id
        String id,
        String worldMapId,
        String name,
        int xPosition,
        int yPosition,
        String markerTypeId,
        String playerArticleId,
        String gmArticleId,
        MapMarkerStatus status,
        Visibility visibility
) {
    public MapMarker(MapMarkerDto mapMarkerDto) {
        this(   null,
                mapMarkerDto.worldMapId(),
                mapMarkerDto.name(),
                mapMarkerDto.xPosition(),
                mapMarkerDto.yPosition(),
                mapMarkerDto.markerTypeId(),
                mapMarkerDto.playerArticleId(),
                mapMarkerDto.gmArticleId(),
                mapMarkerDto.status(),
                mapMarkerDto.visibility()
        );
    }
}
